sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,401 @@
+                table
+                key/value:
+                    "id" : str
+                    "name" : str
+                    columns: list of columns dicts
+                        columns
+                        key/value:
+                            "name": str
+                            "data_type": str
+                            "default": str
+                            "nullable": bool
+                indexes : list of index dicts
+                    index
+                    key/value:
+                        "name": str
+                        "type": str
+                        "is_unique": bool
+                        "is_primary_key": bool
+                        "is_unique_constraint": bool
+                        "is_disabled": bool,
+                        "column_names": str
+                foreign_keys : list of foreign key dicts
+                    foreign_key
+                    key/value:
+                        "foreign_key_name": str
+                        "referencing_table": str
+                        "referencing_column": str
+                        "referenced_table": str
+                        "referenced_column": str
+                partitions: partition dict
+                    partition
+                    key/value:
+                        "partition_count": int
+        """
+        start_time = time.thread_time()
+        self._data_submitter.reset()
+        self._data_submitter.set_base_event_data(
+            self._check.resolved_hostname,
+            self._check.non_internal_tags,
+            self._check._config.cloud_metadata,
+            "{},{}".format(
+                self._check.static_info_cache.get(STATIC_INFO_VERSION, ""),
+                self._check.static_info_cache.get(STATIC_INFO_ENGINE_EDITION, ""),
+            ),
+        )
+
+        databases = self._check.get_databases()
+        db_infos = self._query_db_informations(databases)
+        self._data_submitter.store_db_infos(db_infos)
+
+        @tracked_method(agent_check_getter=agent_check_getter)
+        def fetch_schema_data(cursor, db_name):
+            schemas = self._query_schema_information(cursor)
+            for schema in schemas:
+                tables = self._get_tables(schema, cursor)
+                tables_chunks = list(get_list_chunks(tables, self.TABLES_CHUNK_SIZE))
+                for tables_chunk in tables_chunks:
+                    if time.thread_time() - start_time > self.MAX_EXECUTION_TIME:
+                        # TODO Report truncation to the backend
+                        self._log.warning(
+                            """Truncated data due to the effective execution time reaching {},
+                             stopped on db - {} on schema {}""".format(
+                                self.MAX_EXECUTION_TIME, db_name, schema["name"]
+                            )
+                        )
+                        raise StopIteration
```

Comment:
I suggest we raise with a simple error msg
```suggestion
                        raise StopIteration("some error messages")
```

Reply:
yep, good idea